### PR TITLE
Add slurm driver for scheduler

### LINKED ISSUE #7795: Implement SLURM driver

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**
Resolves https://github.com/equinor/ert/issues/6676

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [x] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/scheduler/slurm_driver.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    FAILED = auto()
+    CANCELED = auto()
+    COMPLETING = auto()
+    CONFIGURING = auto()
+
+
+@dataclass
+class JobData:
+    iens: int
+    submitted_timestamp: float
```
Comment (berland) [inline src/ert/scheduler/slurm_driver.py:38]: I don't think we need `submitted_timestamp` in the slurm driver.

File: src/ert/scheduler/slurm_driver.py [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+    def _submit_cmd(
+        self,
+        name: str = "dummy",
+        runpath: Optional[Path] = None,
+        num_cpu: Optional[int] = 1,
+    ):
+        sbatch_with_args: list[str] = [
+            str(self._sbatch),
+            f"--job-name={name}",
+            f"-D={runpath}",
```
Comment (berland) [inline src/ert/scheduler/slurm_driver.py:116]: we can use --chdir instead of -D to make it more readable.

Comment (eivindjahren) [reply src/ert/scheduler/slurm_driver.py:116]: done

File: src/ert/scheduler/slurm_driver.py (hunk 1 above)
Comment (eivindjahren) [reply src/ert/scheduler/slurm_driver.py:38]: done

**Review by berland — APPROVED** [review]

File: src/ert/scheduler/__init__.py [hunk 3]
```diff
@@ -48,8 +51,28 @@ def create_driver(config: QueueConfig) -> Driver:
...
+            sbatch_cmd=queue_config.get("SBATCH", "sbatch"),
+            scancel_cmd=queue_config.get("SCANCEL", "scancel"),
+            scontrol_cmd=queue_config.get("SCONTROL", "scontrol"),
+            squeue_cmd=queue_config.get("SQUEUE", "squeue"),
+            exclude_hosts=queue_config.get("EXCLUDE_HOST", ""),
+            include_hosts=queue_config.get("INCLUDE_HOST", ""),
+            memory_per_cpu=queue_config.get("MEMORY_PER_CPU"),
+            memory=queue_config.get("MEMORY", ""),
+            max_runtime=queue_config.get("MAX_RUNTIME"),
+            partition=queue_config.get("PARTITION"),
```
[the remark this answers is not in the document]
Comment (eivindjahren) [reply src/ert/scheduler/__init__.py:68]: Fixed

**Pull request merged**